### PR TITLE
UCP/UCT/MD: add diag prints to mem_reg path

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
## What
add diag prints to mem_reg path

## Why ?
for debug purposes

### CODE REVIEW

File: src/uct/base/uct_md.c [hunk 1]
```diff
@@ -444,17 +444,29 @@ uct_md_mem_advise(uct_md_h md, uct_mem_h memh, void *addr, size_t length,
...
+
 ucs_status_t uct_md_mem_reg(uct_md_h md, void *address, size_t length,
                             unsigned flags, uct_mem_h *memh_p)
 {
     ucs_status_t status;
 
     if ((length == 0) || (address == NULL)) {
+        ucs_log(uct_md_reg_log_lvl(flags),
+                "uct_md_mem_reg(): %s: length = %zu, address = %p, "
+                "length and address must be valid",
```
Comment (yosefe) [inline src/uct/base/uct_md.c:460]: `uct_md_mem_reg(address=%p length=%zu): invalid parameters`

File: src/uct/base/uct_md.c [hunk 2]
```diff
@@ -444,17 +444,29 @@ uct_md_mem_advise(uct_md_h md, uct_mem_h memh, void *addr, size_t length,
...
+                ucs_status_string(UCS_ERR_INVALID_PARAM), length, address);
         return UCS_ERR_INVALID_PARAM;
     }
 
     status = uct_mem_check_flags(flags);
     if (status != UCS_OK) {
+        ucs_log(uct_md_reg_log_lvl(flags),
```
Comment (yosefe) [inline src/uct/base/uct_md.c:467]: `uct_md_mem_reg(flags=0x%x): invalid flags`

File: src/uct/ib/base/ib_md.c [hunk 3]
```diff
@@ -801,6 +801,11 @@ static ucs_status_t uct_ib_mem_reg(uct_md_h uct_md, void *address, size_t length
 
     memh = uct_ib_memh_alloc(md);
     if (memh == NULL) {
+        if (!(flags & UCT_MD_MEM_FLAG_HIDE_ERRORS)) {
+            ucs_error("%s: cannot allocate memory handle on md %p",
+                      ucs_status_string(UCS_ERR_NO_MEMORY), md);
```
Comment (yosefe) [inline src/uct/ib/base/ib_md.c:806]: "md %p: failed to allocate memory handle"

**Comment by yosefe** [comment]
@evgeny-leksikov pls also check code style errors

**Comment by evgeny-leksikov** [comment]
> @evgeny-leksikov pls also check code style errors

https://github.com/openucx/ucx/issues/6126#issuecomment-768835389

File: src/uct/ib/base/ib_md.c [hunk 4]
```diff
@@ -801,6 +801,9 @@ static ucs_status_t uct_ib_mem_reg(uct_md_h uct_md, void *address, size_t length
 
     memh = uct_ib_memh_alloc(md);
     if (memh == NULL) {
+        ucs_log((flags & UCT_MD_MEM_FLAG_HIDE_ERRORS) ? UCS_LOG_LEVEL_DIAG :
```
Comment (yosefe) [inline src/uct/ib/base/ib_md.c:804]: maybe uct_md_reg_log_lvl should be helper in uct_md.h, reuse it here?

**Review by yosefe — COMMENTED** [review]
pls see https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=12994&view=logs&jobId=cc064a77-22b5-56bf-ecc0-70b5fe764261&j=cc064a77-22b5-56bf-ecc0-70b5fe764261&t=aedfd754-44a6-53e6-6843-8659d800fee2

File: src/uct/ib/base/ib_md.c (hunk 4 above)
Comment (evgeny-leksikov) [reply src/uct/ib/base/ib_md.c:804]: ok, also thought about this

**Comment by evgeny-leksikov** [comment]
> pls see https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=12994&view=logs&jobId=cc064a77-22b5-56bf-ecc0-70b5fe764261&j=cc064a77-22b5-56bf-ecc0-70b5fe764261&t=aedfd754-44a6-53e6-6843-8659d800fee2

I saw, these are not aligned with our code style, I added comment to the clang issue

File: src/uct/base/uct_md.h [hunk 5]
```diff
@@ -18,6 +18,11 @@
 #include <string.h>
 
 
+#define uct_md_reg_log_lvl(_flags, _fmt, ...) \
```
Comment (yosefe) [inline src/uct/base/uct_md.h:21]: uct_md_log_mem_reg_error

**Review by yosefe — APPROVED** [review]

**Pull request merged**